### ISSUE TITLE
calculate_slope leaking DEM `nan` `nodata` values to slope output

### ISSUE THREAD

**Issue by dcdenu4** [natcap/pygeoprocessing#320]
A user on the [forums](community.naturalcapitalproject.org/t/sdr-error-exception-encountered-in-raster-calculator/3715) was experience an error in the `sdr._calculate_ls_factor` function.

```python
Traceback (most recent call last):
  File "C:\Users\ddenu\Workspace\NatCap\User-Data\SDR error_ exception encountered in raster_calculator\sdr_temp.py", line 144, in <module>
    calculate_ls_factor(flow_accum_path, slope_path, weighted_avg_aspect_path, l_max, ls_path)
  File "C:\Users\ddenu\Workspace\NatCap\User-Data\SDR error_ exception encountered in raster_calculator\sdr_temp.py", line 130, in calculate_ls_factor
    pygeoprocessing.raster_calculator(
  File "C:\Users\ddenu\Workspace\NatCap\Repositories\venv-space\invest-debug\lib\site-packages\pygeoprocessing\geoprocessing.py", line 443, in raster_calculator
    target_block = local_op(*data_blocks)
  File "C:\Users\ddenu\Workspace\NatCap\User-Data\SDR error_ exception encountered in raster_calculator\sdr_temp.py", line 115, in ls_factor_function
    m_exp[~big_slope_mask] = m_table[m_indexes]
IndexError: index 4 is out of bounds for axis 0 with size 4
```

I traced this back to the slope raster input to that function having `nan` values. The users DEM input has a `nodata` value of `nan` but the raster itself doesn't actually have any `nodata` values. It doesn't appear to me that `calculate_slope` safe guards against `nan` `nodata` values for the input DEM. It checks if `nodata` is `None` but [otherwise initializes](https://github.com/natcap/pygeoprocessing/blob/55fbbe9ef867308e26be71356b22e98059c2f897/src/pygeoprocessing/geoprocessing_core.pyx#LL384C4-L387C34) `dem_array` to `nan`. 

My best guess so far is `nan` is leaking in from `dem_array` and `slope_array` being buffered, which would allow `nan` values to propagate.

**Comment by phargogh** [comment]
@dcdenu4, we just merged @emlys's patch in #356 to handle `nan` nodata values in the slope calculations.  Do you think that fix will have addressed this issue?

**Comment by dcdenu4** [comment]
Thanks for flagging @phargogh. Yes, I think this should catch any `nan` nodata values from the DEM and set them to `slope_nodata` for the output.

Should be fixed in #356
